include/navigation/perfect_memory.h

```diff
@@ -115,29 +103,41 @@ class PerfectMemory : public VisualNavigationBase
         return {0, getNumSnapshots()};
     }
 
+    //! Get the resolution of images
+    const cv::Size &getUnwrapResolution() const { return m_UnwrapRes; }
+
 protected:
+    static const Window defaultWindow;
```

Comment:
Minor nit: but I feel like this should probably be public as it might be handy for users.

I guess you could probably also make it ``constexpr``.

Reply:
this was a bit of a gross hack around not being able to initialise default parameters with a non-static method - we don't really want users initialising windows to this, they should use ``fullWindow()`` instead. Sadly wouldn't we need C++ 17 for inline constants rather than gross static initialisation syntax?

Reply:
True: you would need a separate definition for ``defaultWindow`` somewhere. So it's probably not worth it. Let's just make users use ``fullWindow()``.

Reply:
actually, the whole ``defaultWindow`` thing was gross - checkout my last commit